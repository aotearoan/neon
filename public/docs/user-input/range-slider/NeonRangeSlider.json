{
  "displayName": "NeonRangeSlider",
  "exportName": "default",
  "description": "",
  "tags": {},
  "props": [
    {
      "name": "modelValue",
      "description": "This is the <em>v-model</em> property which is an array containing the lower and upper bounds of the selected range.",
      "type": {
        "name": "Array as () => Array<number>"
      },
      "required": true
    },
    {
      "name": "ids",
      "description": "The list of ids for the lower bound and upper bound inputs, e.g. ['lowerBoundId', 'upperBoundId']",
      "type": {
        "name": "Array as () => Array<string>"
      },
      "required": false
    },
    {
      "name": "color",
      "description": "Slider color.",
      "type": {
        "name": "String as () => NeonFunctionalColor"
      },
      "defaultValue": {
        "func": false,
        "value": "NeonFunctionalColor.LowContrast"
      }
    },
    {
      "name": "output",
      "description": "Disable output display if set to false",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "true"
      }
    },
    {
      "name": "legend",
      "description": "Disable legend if set to false",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "true"
      }
    },
    {
      "name": "tooltip",
      "description": "Disable tooltip if set to false",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "true"
      }
    },
    {
      "name": "step",
      "description": "The size of steps between values the user can select. Defaults to 1 unless percentage = true in which case it will\ndefault to 0.01.",
      "type": {
        "name": "number"
      },
      "required": false
    },
    {
      "name": "decimals",
      "description": "The rounding precision for display purposes",
      "type": {
        "name": "number"
      },
      "required": false
    },
    {
      "name": "valueTemplate",
      "description": "A format template string used for display purposes. Use the placeholder {value} to reference the value in the\nformat string. e.g. value = 90, ${value} => $90",
      "type": {
        "name": "string"
      },
      "required": false
    },
    {
      "name": "disableFormatting",
      "description": "Disable formatting, e.g. in the case of a year value -> display as 2020, not 2,020.",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "false"
      }
    },
    {
      "name": "percentage",
      "description": "Automatically applies % formatting, e.g. if the value = 0.15 it will be displayed as 15%",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "false"
      }
    },
    {
      "name": "min",
      "description": "The minimum range value",
      "type": {
        "name": "number"
      },
      "defaultValue": {
        "func": false,
        "value": "0"
      }
    },
    {
      "name": "max",
      "description": "The maximum range value. The default value is 100 except when percentage = true the default is 1 (100%).",
      "type": {
        "name": "number"
      },
      "required": false
    },
    {
      "name": "disabled",
      "description": "Component disabled state.",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "false"
      }
    },
    {
      "name": "lowerBoundLabel",
      "description": "ARIA label for the lower bound slider.",
      "type": {
        "name": "string"
      },
      "defaultValue": {
        "func": false,
        "value": "'Lower bound'"
      }
    },
    {
      "name": "upperBoundLabel",
      "description": "ARIA label for the upper bound slider.",
      "type": {
        "name": "string"
      },
      "defaultValue": {
        "func": false,
        "value": "'Upper bound'"
      }
    }
  ],
  "events": [
    {
      "name": "update:modelValue",
      "description": "Event triggered when the lower or upper bounds of the value change.",
      "type": {
        "names": [
          "number[]"
        ]
      }
    }
  ]
}