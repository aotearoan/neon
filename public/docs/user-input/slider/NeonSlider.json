{
  "displayName": "NeonSlider",
  "description": "<p>\n  The <strong>NeonSlider</strong> component is the equivalent of an <strong>&lt;input type=\"range\" /&gt;</strong>. In\n  addition to this, there is support for automatic formatting the min, max and selected value plus options for\n  percentage formatting and providing custom templates, e.g. currency formatting.\n</p>\n<p><strong>NeonNumber</strong> also supports all relevant properties found on an HTML &lt;input&gt;.</p>",
  "tags": {},
  "exportName": "default",
  "props": [
    {
      "name": "modelValue",
      "description": "The current input <em>value</em>.",
      "type": {
        "name": "number"
      },
      "required": true
    },
    {
      "name": "id",
      "description": "Id of the range input.",
      "type": {
        "name": "string"
      },
      "required": false
    },
    {
      "name": "color",
      "description": "Slider color.",
      "type": {
        "name": "String as () => NeonFunctionalColor"
      },
      "defaultValue": {
        "func": false,
        "value": "NeonFunctionalColor.LowContrast"
      }
    },
    {
      "name": "output",
      "description": "Show/hide the output.",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "true"
      }
    },
    {
      "name": "legend",
      "description": "Show/hide the legend.",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "true"
      }
    },
    {
      "name": "tooltip",
      "description": "Show/hide the tooltip.",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "true"
      }
    },
    {
      "name": "percentage",
      "description": "Automatically applies % formatting, e.g. if the value = 0.15 it will be displayed as 15%.",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "false"
      }
    },
    {
      "name": "step",
      "description": "The size of steps between values the user can select. The default value is 1 except when percentage = true the\ndefault is 0.01 (1%).",
      "type": {
        "name": "number"
      },
      "required": false
    },
    {
      "name": "decimals",
      "description": "The rounding precision for display purposes.",
      "type": {
        "name": "number"
      },
      "required": false
    },
    {
      "name": "valueTemplate",
      "description": "A template string used for formatting the display value. Use the placeholder {value} to reference the value in the\ntemplate string. e.g. value = 90, ${value} => $90.",
      "type": {
        "name": "string"
      },
      "required": false
    },
    {
      "name": "disableFormatting",
      "description": "Disable formatting, e.g. in the case of a year value -> display as 2020, not 2,020.",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "false"
      }
    },
    {
      "name": "min",
      "description": "The minimum range value.",
      "type": {
        "name": "number"
      },
      "defaultValue": {
        "func": false,
        "value": "0"
      }
    },
    {
      "name": "max",
      "description": "The maximum range value. The default value is 100 except when percentage = true the default is 1 (100%).",
      "type": {
        "name": "number"
      },
      "required": false
    },
    {
      "name": "lowerBound",
      "description": "The lower bound within the range of values.",
      "tags": {
        "ignore": [
          {
            "description": true,
            "title": "ignore"
          }
        ]
      },
      "type": {
        "name": "number"
      },
      "required": false
    },
    {
      "name": "upperBound",
      "description": "The upper bound within the range of values.",
      "tags": {
        "ignore": [
          {
            "description": true,
            "title": "ignore"
          }
        ]
      },
      "type": {
        "name": "number"
      },
      "required": false
    },
    {
      "name": "disabled",
      "description": "Component disabled state.",
      "type": {
        "name": "boolean"
      },
      "defaultValue": {
        "func": false,
        "value": "false"
      }
    }
  ],
  "events": [
    {
      "name": "update:modelValue",
      "description": "Event triggered when the value changes.",
      "type": {
        "names": [
          "number"
        ]
      }
    }
  ]
}