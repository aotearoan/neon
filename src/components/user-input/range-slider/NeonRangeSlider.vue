<template>
  <div
    class="neon-range-slider"
    :class="{
      'neon-range-slider--disabled': disabled,
      'neon-range-slider--all-upper-bound': modelValue[0] === modelValue[1] && modelValue[0] === max,
    }"
  >
    <output class="neon-slider__output" v-if="output">
      <span class="neon-range-slider__output-range">
        <span>{{ formattedValues[0] }}</span>
        <span class="neon-range-slider__output-range-separator"> : </span>
        <span>{{ formattedValues[1] }}</span>
      </span>
    </output>
    <neon-slider
      :id="ids && ids[0]"
      :color="color"
      :aria-label="lowerBoundLabel"
      :modelValue="modelValue[0]"
      @update:modelValue="changeLowerBound"
      :upperBound="modelValue[1]"
      :disabled="disabled"
      :min="min"
      :max="max"
      :step="step"
      :decimals="decimals"
      :value-template="valueTemplate"
      :disableFormatting="disableFormatting"
      :percentage="percentage"
      :tooltip="tooltip"
      :legend="legend"
      class="neon-range-slider__lower"
      :output="false"
    ></neon-slider>
    <neon-slider
      :id="ids && ids[1]"
      :color="color"
      :aria-label="upperBoundLabel"
      :modelValue="modelValue[1]"
      @update:modelValue="changeUpperBound"
      :lowerBound="modelValue[0]"
      :disabled="disabled"
      :min="min"
      :max="max"
      :step="step"
      :decimals="decimals"
      :value-template="valueTemplate"
      :disableFormatting="disableFormatting"
      :percentage="percentage"
      :tooltip="tooltip"
      :legend="false"
      class="neon-range-slider__upper"
      :output="false"
    ></neon-slider>
  </div>
</template>

<script lang="ts" src="./NeonRangeSlider.ts"></script>
