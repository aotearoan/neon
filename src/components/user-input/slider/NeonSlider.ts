import { computed, defineComponent, useAttrs } from 'vue';
import { NeonFunctionalColor } from '@/common/enums/NeonFunctionalColor';
import { NeonNumberUtils } from '@/common/utils/NeonNumberUtils';

/**
 * <p>
 *   The <strong>NeonSlider</strong> component is the equivalent of an <strong>&lt;input type="range" /&gt;</strong>. In
 *   addition to this, there is support for automatic formatting the min, max and selected value plus options for
 *   percentage formatting and providing custom templates, e.g. currency formatting.
 * </p>
 * <p><strong>NeonNumber</strong> also supports all relevant properties found on an HTML &lt;input&gt;.</p>
 */
export default defineComponent({
  name: 'NeonSlider',
  props: {
    /**
     * The current input <em>value</em>.
     */
    modelValue: { type: Number, required: true },
    /**
     * id of the range input.
     */
    id: { type: String, required: false },
    /**
     * Slider color.
     */
    color: { type: String as () => NeonFunctionalColor, default: NeonFunctionalColor.LowContrast },
    /**
     * The locale used for display purposes. This defaults to the browser's locale if none is provided.
     */
    locale: { type: String, default: null },
    /**
     * Show/hide the output.
     */
    output: { type: Boolean, default: true },
    /**
     * Show/hide the legend.
     */
    legend: { type: Boolean, default: true },
    /**
     * Show/hide the tooltip.
     */
    tooltip: { type: Boolean, default: true },
    /**
     * Automatically applies % formatting, e.g. if the value = 0.15 it will be displayed as 15%.
     */
    percentage: { type: Boolean, default: false },
    /**
     * The size of steps between values the user can select. The default value is 1 except when percentage = true the
     * default is 0.01 (1%).
     */
    step: { type: Number, required: false },
    /**
     * The rounding precision for display purposes.
     */
    decimals: { type: Number, required: false },
    /**
     * A template string used for formatting the display value. Use the placeholder {value} to reference the value in the
     * template string. e.g. value = 90, ${value} => $90.
     */
    valueTemplate: { type: String, required: false },
    /**
     * Disable formatting, e.g. in the case of a year value -> display as 2020, not 2,020.
     */
    disableFormatting: { type: Boolean, default: false },
    /**
     * The minimum range value.
     */
    min: { type: Number, default: 0 },
    /**
     * The maximum range value. The default value is 100 except when percentage = true the default is 1 (100%).
     */
    max: { type: Number, required: false },
    /**
     * The lower bound within the range of values.
     * @ignore
     */
    lowerBound: { type: Number, required: false },
    /**
     * The upper bound within the range of values.
     * @ignore
     */
    upperBound: { type: Number, required: false },
    /**
     * Component disabled state.
     */
    disabled: { type: Boolean, default: false },
  },
  emits: [
    /**
     * Event triggered when the value changes.
     *
     * @type {number} the raw selected numeric value.
     */
    'update:modelValue',
  ],
  setup(props, { emit }) {
    const attrs = useAttrs();

    const sanitizedAttributes = computed(() => {
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const { ...sanitized } = attrs;
      return sanitized;
    });

    const formatNumber = (value: number) => {
      return !props.disableFormatting
        ? NeonNumberUtils.formatNumber(
            value,
            {
              decimals: props.decimals,
              format: props.valueTemplate,
              percentage: props.percentage,
            },
            props.locale,
          )
        : value;
    };

    const formattedMin = computed(() => formatNumber(props.min));
    const computedMax = computed(() => (props.max !== undefined ? props.max : props.percentage ? 1 : 100));
    const formattedMax = computed(() => formatNumber(computedMax.value));
    const formattedValue = computed(() => formatNumber(props.modelValue));
    const computedStep = computed(() => (props.step !== undefined ? props.step : props.percentage ? 0.01 : 1));

    const changeValue = (event: Event) => {
      const input = event.target as HTMLInputElement;
      let newValue = +input.value;
      // adjust for bounds if set
      if (props.lowerBound !== undefined && newValue < props.lowerBound) {
        newValue = props.lowerBound;
      } else if (props.upperBound !== undefined && newValue > props.upperBound) {
        newValue = props.upperBound;
      }
      emit('update:modelValue', newValue);
    };

    return {
      sanitizedAttributes,
      formattedMin,
      formattedMax,
      formattedValue,
      computedMax,
      computedStep,
      changeValue,
    };
  },
});
