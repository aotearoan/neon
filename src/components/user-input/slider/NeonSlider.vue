<template>
  <div
    class="neon-slider"
    :class="[
      {
        'neon-slider--disabled': disabled,
        'neon-slider--no-legend': !legend,
      },
      `neon-slider--${color}`,
    ]"
    :style="'--min: ' + min + ';--max: ' + computedMax + ';--val: ' + modelValue"
  >
    <output v-if="output" :for="id" class="neon-slider__output">{{ formattedValue }}</output>
    <input
      :id="id"
      type="range"
      :value="modelValue"
      :step="computedStep"
      :min="min"
      :max="computedMax"
      :aria-valuemin="min"
      :aria-valuemax="computedMax"
      :aria-valuenow="modelValue"
      :disabled="disabled"
      @input="changeValue"
      class="neon-slider__input"
      v-bind="sanitizedAttributes"
    />
    <div v-if="legend" class="neon-slider__legend">
      <span class="neon-slider__legend-item">{{ formattedMin }}</span>
      <span class="neon-slider__legend-item">{{ formattedMax }}</span>
    </div>
    <output v-if="tooltip" :for="id" class="neon-slider__tooltip">
      <div class="neon-slider__tooltip-content">
        <span class="neon-slider__tooltip-value">{{ formattedValue }}</span>
      </div>
    </output>
  </div>
</template>

<script lang="ts" src="./NeonSlider.ts"></script>
