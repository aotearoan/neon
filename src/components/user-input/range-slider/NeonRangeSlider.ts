import { computed, defineComponent } from 'vue';
import { NeonNumberUtils } from '@/common/utils/NeonNumberUtils';
import NeonSlider from '@/components/user-input/slider/NeonSlider.vue';
import { NeonFunctionalColor } from '@/common/enums/NeonFunctionalColor';

/**
 *
 */
export default defineComponent({
  name: 'NeonRangeSlider',
  components: {
    NeonSlider,
  },
  props: {
    /**
     * This is the <em>v-model</em> property which is an array containing the lower and upper bounds of the selected range.
     */
    modelValue: { type: Array as () => Array<number>, required: true },
    /**
     * The list of ids for the lower bound and upper bound inputs, e.g. ['lowerBoundId', 'upperBoundId']
     */
    ids: { type: Array as () => Array<string>, required: false },

    /**
     * Slider color.
     */
    color: { type: String as () => NeonFunctionalColor, default: NeonFunctionalColor.LowContrast },
    /**
     * Disable output display if set to false
     */
    output: { type: Boolean, default: true },
    /**
     * Disable legend if set to false
     */
    legend: { type: Boolean, default: true },
    /**
     * Disable tooltip if set to false
     */
    tooltip: { type: Boolean, default: true },
    /**
     * The size of steps between values the user can select. Defaults to 1 unless percentage = true in which case it will
     * default to 0.01.
     */
    step: { type: Number, required: false },
    /**
     * The rounding precision for display purposes
     */
    decimals: { type: Number, required: false },
    /**
     * A format template string used for display purposes. Use the placeholder {value} to reference the value in the
     * format string. e.g. value = 90, ${value} => $90
     */
    valueTemplate: { type: String, required: false },
    /**
     * Disable formatting, e.g. in the case of a year value -> display as 2020, not 2,020.
     */
    disableFormatting: { type: Boolean, default: false },
    /**
     * Automatically applies % formatting, e.g. if the value = 0.15 it will be displayed as 15%
     */
    percentage: { type: Boolean, default: false },
    /**
     * The minimum range value
     */
    min: { type: Number, default: 0 },
    /**
     * The maximum range value. The default value is 100 except when percentage = true the default is 1 (100%).
     */
    max: { type: Number, required: false },
    /**
     * Component disabled state.
     */
    disabled: { type: Boolean, default: false },
    /**
     * ARIA label for the lower bound slider.
     */
    lowerBoundLabel: { type: String, default: 'Lower bound' },
    /**
     * ARIA label for the upper bound slider.
     */
    upperBoundLabel: { type: String, default: 'Upper bound' },
  },
  emits: [
    /**
     * Event triggered when the lower or upper bounds of the value change.
     *
     * @type {number[]} An array containing the raw numeric upper and lower bounds of the selection.
     */
    'update:modelValue',
  ],
  setup(props, { emit }) {
    const formattedValues = computed((): string[] => {
      const options = {
        decimals: props.decimals,
        format: props.valueTemplate,
        percentage: props.percentage,
      };

      return !props.disableFormatting
        ? [
            NeonNumberUtils.formatNumber(props.modelValue[0], options),
            NeonNumberUtils.formatNumber(props.modelValue[1], options),
          ]
        : [`${props.modelValue[0]}`, `${props.modelValue[1]}`];
    });

    const emitValues = (values: number[]) => {
      emit('update:modelValue', values);
    };

    const changeLowerBound = (value: number) => {
      const values = props.modelValue.map((v) => v);
      values[0] = +value;
      emitValues(values);
    };

    const changeUpperBound = (value: number) => {
      const values = props.modelValue.map((v) => v);
      values[1] = +value;
      emitValues(values);
    };

    return {
      formattedValues,
      changeLowerBound,
      changeUpperBound,
      emitValues,
    };
  },
});
