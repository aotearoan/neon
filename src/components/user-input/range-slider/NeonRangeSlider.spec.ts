import Vue from 'vue';
import { mount } from '@vue/test-utils';
import NeonRangeSlider from './NeonRangeSlider.vue';
import NeonSlider from '../slider/NeonSlider.vue';

Vue.component('NeonSlider', NeonSlider);

describe('NeonRangeSlider', () => {
  it('renders enabled', () => {
    // given
    const value = [5, 10];
    const wrapper = mount(NeonRangeSlider, {
      propsData: { value },
      mocks: {
        $t: (key: string) => key,
      },
    });
    // when / then
    expect(wrapper.find('.neon-range-slider--disabled').element).toBeUndefined();
    expect(wrapper.findAll('.neon-slider--disabled').length).toEqual(0);
  });

  it('renders disabled', () => {
    // given
    const value = [5, 10];
    const wrapper = mount(NeonRangeSlider, {
      propsData: { value, disabled: true },
      mocks: {
        $t: (key: string) => key,
      },
    });
    // when / then
    expect(wrapper.find('.neon-range-slider--disabled').element).toBeDefined();
    expect(wrapper.findAll('.neon-slider--disabled').length).toEqual(2);
  });

  it('renders output', () => {
    // given
    const value = [5000, 10000];
    const wrapper = mount(NeonRangeSlider, {
      propsData: { value },
      mocks: {
        $t: (key: string) => key,
      },
    });
    // when / then
    expect(wrapper.find('.neon-range-slider__output-range').text()).toEqual('5,000 : 10,000');
  });

  it('renders output disabled formatting', () => {
    // given
    const value = [5000, 10000];
    const wrapper = mount(NeonRangeSlider, {
      propsData: { value, disableFormatting: true },
      mocks: {
        $t: (key: string) => key,
      },
    });
    // when / then
    expect(wrapper.find('.neon-range-slider__output-range').text()).toEqual('5000 : 10000');
  });

  it('does not render output', () => {
    // given
    const value = [5000, 10000];
    const wrapper = mount(NeonRangeSlider, {
      propsData: { value, output: false },
      mocks: {
        $t: (key: string) => key,
      },
    });
    // when / then
    expect(wrapper.find('.neon-range-slider__output').element).toBeUndefined();
  });

  it('renders all upper bound', () => {
    // given
    const value = [10000, 10000];
    const max = 10000;
    const wrapper = mount(NeonRangeSlider, {
      propsData: { value, max },
      mocks: {
        $t: (key: string) => key,
      },
    });
    // when / then
    expect(wrapper.find('.neon-range-slider--all-upper-bound').element).toBeDefined();
  });

  it('matches snapshot', () => {
    // given
    const value = [5000, 10000];
    const min = 1000;
    const lowerBound = 2000;
    const max = 50000;
    const upperBound = 49000;
    const step = 100;
    const decimals = 2;
    // eslint-disable-next-line no-template-curly-in-string
    const valueTemplate = '${value}';
    const wrapper = mount(NeonRangeSlider, {
      propsData: { value, min, max, lowerBound, upperBound, step, decimals, valueTemplate },
      mocks: {
        $t: (key: string) => key,
      },
    });
    // when / then
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('emits lower change inside bound', () => {
    // given
    const value = [5, 10];
    const newValue = 6;
    const wrapper = mount(NeonRangeSlider, {
      propsData: { value },
      mocks: {
        $t: (key: string) => key,
      },
    });
    // when
    wrapper.find('.neon-range-slider__lower .neon-slider__input').setValue(newValue);
    // then
    expect(wrapper.emitted().input[0]).toEqual([[newValue, value[1]]]);
  });

  it('emits lower change outside bound', () => {
    // given
    const value = [5, 10];
    const newValue = 11;
    const wrapper = mount(NeonRangeSlider, {
      propsData: { value },
      mocks: {
        $t: (key: string) => key,
      },
    });
    // when
    wrapper.find('.neon-range-slider__lower .neon-slider__input').setValue(newValue);
    // then
    expect(wrapper.emitted().input[0]).toEqual([[value[1], value[1]]]);
  });

  it('emits upper change outside bound', () => {
    // given
    const value = [5, 10];
    const newValue = 4;
    const wrapper = mount(NeonRangeSlider, {
      propsData: { value },
      mocks: {
        $t: (key: string) => key,
      },
    });
    // when
    wrapper.find('.neon-range-slider__upper .neon-slider__input').setValue(newValue);
    // then
    expect(wrapper.emitted().input[0]).toEqual([[value[0], value[0]]]);
  });

  it('emits upperBound change inside bound', () => {
    // given
    const value = [5, 10];
    const newValue = 6;
    const wrapper = mount(NeonRangeSlider, {
      propsData: { value },
      mocks: {
        $t: (key: string) => key,
      },
    });
    // when
    wrapper.find('.neon-range-slider__upper .neon-slider__input').setValue(newValue);
    // then
    expect(wrapper.emitted().input[0]).toEqual([[value[0], newValue]]);
  });
});
