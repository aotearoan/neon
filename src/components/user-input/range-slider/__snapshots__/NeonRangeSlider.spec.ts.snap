// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeonRangeSlider matches snapshot 1`] = `
<div class="neon-range-slider" lowerBound="2000" upperBound="49000"><output class="neon-slider__output"><span class="neon-range-slider__output-range"><span>$5,000.00</span><span class="neon-range-slider__output-range-separator"> : </span><span>$10,000.00</span></span></output>
  <div class="neon-slider neon-slider--low-contrast neon-range-slider__lower" style="--min: 1000; --max: 50000; --val: 5000;" aria-label="Lower bound">
    <!--v-if--><input type="range" step="100" min="1000" max="50000" aria-valuemin="1000" aria-valuemax="50000" aria-valuenow="5000" disabled="false" class="neon-slider__input neon-range-slider__lower" aria-label="Lower bound" value="5000"></input>
    <div class="neon-slider__legend"><span class="neon-slider__legend-item">$1,000.00</span><span class="neon-slider__legend-item">$50,000.00</span></div><output class="neon-slider__tooltip">
      <div class="neon-slider__tooltip-content"><span class="neon-slider__tooltip-value">$5,000.00</span></div>
    </output>
  </div>
  <div class="neon-slider neon-slider--no-legend neon-slider--low-contrast neon-range-slider__upper" style="--min: 1000; --max: 50000; --val: 10000;" aria-label="Upper bound">
    <!--v-if--><input type="range" step="100" min="1000" max="50000" aria-valuemin="1000" aria-valuemax="50000" aria-valuenow="10000" disabled="false" class="neon-slider__input neon-range-slider__upper" aria-label="Upper bound" value="10000"></input>
    <!--v-if--><output class="neon-slider__tooltip">
      <div class="neon-slider__tooltip-content"><span class="neon-slider__tooltip-value">$10,000.00</span></div>
    </output>
  </div>
</div>
`;
